pydra/engine/helpers_file.py

```diff
@@ -544,31 +51,40 @@ def ensure_list(filename):
     return None
 
 
-# not sure if this might be useful for Function Task
-def copyfile_input(inputs, output_dir):
-    """Implement the base class method."""
-    from .specs import attr_fields, File, MultiInputFile
-
-    map_copyfiles = {}
-    for fld in attr_fields(inputs):
-        copy = fld.metadata.get("copyfile")
-        if copy is not None and fld.type not in [File, MultiInputFile]:
-            raise Exception(
-                f"if copyfile set, field has to be a File " f"but {fld.type} provided"
-            )
-        file = getattr(inputs, fld.name)
-        if copy in [True, False] and file != attr.NOTHING:
-            if isinstance(file, list):
-                map_copyfiles[fld.name] = []
-                for el in file:
-                    newfile = output_dir.joinpath(Path(el).name)
-                    copyfile(el, newfile, copy=copy)
-                    map_copyfiles[fld.name].append(str(newfile))
-            else:
-                newfile = output_dir.joinpath(Path(file).name)
-                copyfile(file, newfile, copy=copy)
-                map_copyfiles[fld.name] = str(newfile)
-    return map_copyfiles or None
+def copy_nested_files(
```

Comment:
perhaps we could have tests for these new functions in test_helpers_file?

Reply:
Good idea. I have added them now (and found/fixed a bug in the process)